### ISSUE TITLE
ValueError or incorrect result returned for HEOS::Air for Hmass and SMass inputs 

### ISSUE THREAD

**Issue by tjmackman** [CoolProp/CoolProp#2022]
ValueError or bad value returned for pressure for HEOS::Air when using Hmass and Smass inputs.  Tested in CoolProp 6.1.0 and 6.4.1, Windows 8.1, via Python interface (python 3.5 and python 3.7).
 
### Code to reproduce
```
import CoolProp.CoolProp as cp
# comment/uncomment each line as applicable
cp.PropsSI("P","Hmass",429667.3064,"Smass",2735.612747,"HEOS::Air") # ValueError: p is not a valid number
cp.PropsSI("P","Hmass",430867.4193,"Smass",2732.980272,"HEOS::Air") # ValueError: p is not a valid number
cp.PropsSI("P","Hmass",427500.0,"Smass",2740.0,"HEOS::Air")         # ValueError: p is not a valid number
cp.PropsSI("P","Hmass",435000.0,"Smass",2654.0,"HEOS::Air")         # bad value
cp.PropsSI("P","Hmass",435000.0,"Smass",2724.0,"HEOS::Air")         # ValueError: p is not a valid number
```

Calling with REFPROP::Air (version 9.1, or 10) produces the expected result in each case.

**Comment by ibell** [comment]
Yeah, HS inputs are tricky, they are, believe it or not, much more reliable than they used to be.  What application are you working with that you are using HS inputs (I put together some new HS routines for the ORC 2021 conference....)  Can you use TS inputs or something else other than HS which is pretty much always the worst/slowest?

**Comment by tjmackman** [comment]
Ok, thanks for this.  The application is a generic routine that determines static p, T from input mass flow, area, and stagnation p0, T0 for pipe flows.  Iterating from an initial guess of static enthalpy and known fixed entropy is convenient, as it means no assumptions about the equation of state are needed, so the routine is valid for real gas.  We have no problem in this case (call REFPROP, which is just a bit slower, or use isentropic relations / perfect gas assumption / non-HS fluid properties calls as it's air), and actually this is the first time I've noticed the HS inputs being unstable, so it's not too bad :)  I think there are probably a few ways to do what we want, or ultimately we can look at using a faster low level hook into REFPROP.

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
